scripts/machine-launch-script.sh

```diff
@@ -1,85 +1,271 @@
+	exit 1
+    fi
+
+    OS_FLAVOR=$(grep '^ID=' /etc/os-release | awk -F= '{print $2}' | tr -d '"')
+    OS_VERSION=$(grep '^VERSION_ID=' /etc/os-release | awk -F= '{print $2}' | tr -d '"')
+
+    echo "machine launch script started" > machine-launchstatus
+    chmod ugo+r machine-launchstatus
+
+    # platform-specific setup 
+    case "$OS_FLAVOR" in
+	ubuntu)
+	    ;;
+	centos)
+	    ;;
+	*)
+	    echo "::WARNING:: Unknown OS flavor '$OS_FLAVOR'. Unable to do platform-specific setup."
+	    ;;
+    esac
+
+
+    # everything else is platform-agnostic and could easily be expanded to Windows and/or OSX
+
+    SUDO=""
+    prefix_parent=$(dirname "$CONDA_INSTALL_PREFIX")
+    if [[ ! -e "$prefix_parent" ]]; then
+	mkdir -p "$prefix_parent" || SUDO=sudo
+    elif [[ ! -w "$prefix_parent" ]]; then
+	SUDO=sudo
+    fi
+
+    if [[ -n "$SUDO" ]]; then
+	echo "::INFO:: using 'sudo' to install conda"
+	# ensure files are read-execute for everyone
+	umask 022
+    fi
+
+    # to enable use of sudo and avoid modifying 'secure_path' in /etc/sudoers, we specify the full path to conda
+    CONDA_EXE="${CONDA_INSTALL_PREFIX}/bin/conda"
+
+    if [[ -x "$CONDA_EXE" ]]; then
+	echo "::INFO:: '$CONDA_EXE' already exists, skipping conda install"
+    else
+	wget -O install_conda.sh "$CONDA_INSTALLER"  || curl -fsSLo install_conda.sh "$CONDA_INSTALLER"
+	echo "::INFO:: installing conda to '$CONDA_INSTALL_PREFIX'"
+	# -b for non-interactive install
+	$SUDO bash ./install_conda.sh -b -p "$CONDA_INSTALL_PREFIX"
+
+        # see https://conda-forge.org/docs/user/tipsandtricks.html#multiple-channels
+        # for more information on strict channel_priority
+        $SUDO "$CONDA_EXE" config --system --set channel_priority strict
+        # By default, don't mess with people's PS1, I personally find it annoying
+        $SUDO "$CONDA_EXE" config --system --set changeps1 false
+
+	conda_init_extra_args=()
+	if [[ -n "$SUDO"  || "$USER" == root ]]; then
+	    # if we're installing into a root-owned directory using sudo, or we're already root
+	    # initialize conda in the system-wide rcfiles
+	    conda_init_extra_args=(--no-user --system)
+	fi
+	$SUDO "${CONDA_EXE}" init $DRY_RUN_OPTION "${conda_init_extra_args[@]}" bash
+    fi
+
+    # https://conda-forge.org/feedstock-outputs/ 
+    #	filterable list of all conda-forge packages
+    # https://conda-forge.org/#contribute 
+    #	instructions on adding a recipe
+    # https://docs.conda.io/projects/conda/en/latest/user-guide/concepts/pkg-specs.html#package-match-specifications
+    #   documentation on package_spec syntax
+    CONDA_PACKAGE_SPECS=()
+
+
+    # https://docs.conda.io/projects/conda-build/en/latest/resources/compiler-tools.html#using-the-compiler-packages
+    #    for notes on using the conda compilers
+    CONDA_PACKAGE_SPECS=( gcc_linux-64 gxx_linux-64 )
+
+    # if we want to install prebuilt riscv toolchain, the bdutrosv channel has riscv-gnu-toolchain
+
```

Comment:
I've been thinking that it would be really nice to have some more handy shell utilities installed that weren't part of what we were installing before.  My list so far of stuff I like to have around is:
* jq
* shyaml
* ripgrep
* shellcheck

But I didn't want to start throwing in more stuff that I like to have.  It's easy for me to just add them.

Reply:
I'm fine adding these as long as they 1. don't take too much extra time to add and 2. they are clearly marked as optional.

Reply:
I think until and unless we're using them in scripts, I'll keep them out of the script.

Reply:
My preference is to limit this to only required dependencies, but if they are clearly labelled as @abejgonzalez suggested, it's fine.